iceoryx_posh/test/moduletests/test_posh_runtime.cpp

```diff
@@ -493,3 +493,41 @@ TEST_F(PoshRuntime_test, CreateRunnableReturnValue)
     /// @todo I am passing runnableDeviceIdentifier as 1, but it returns 0, is this expected?
     // EXPECT_EQ(runnableDeviceIdentifier, runableData->m_runnableDeviceIdentifier);
 }
+
+class PoshRuntimeTestAccess : public PoshRuntime
+{
+  public:
+    using PoshRuntime::factory_t;
+    using PoshRuntime::setRuntimeFactory;
+
+    PoshRuntimeTestAccess(const iox::ProcessName_t& s)
+        : PoshRuntime(s)
+    {
+    }
+};
+
+TEST_F(PoshRuntime_test, SetValidRuntimeFactorySucceeds)
+{
+    bool callbackWasCalled = false;
+    iox::cxx::optional<PoshRuntimeTestAccess> runtime;
+    PoshRuntimeTestAccess::factory_t factory = [&](const iox::ProcessName_t& name) -> PoshRuntime& {
+        callbackWasCalled = true;
+        if (!runtime.has_value())
+        {
+            runtime.emplace(name);
+        }
+        return runtime.value();
+    };
+
+    PoshRuntimeTestAccess::setRuntimeFactory(factory);
+    PoshRuntimeTestAccess::getInstance("/instance");
+
```

Comment:
It should also be checked whether the provided runtime factory was set and not something else. This could be done by adding a side effect (e.g. a static counter) to the PoshRunTimeTestAccess class and checking the counter value after getInstance was called.

Please add this check if possible.

Reply:
This is actually already tested with the flag. Nevermind.